app/scripts/directives/addMobileResource.js

```diff
@@ -0,0 +1,33 @@
+'use strict';
+
+(function () {
+  angular.module('openshiftConsole').component('addMobileResource', {
+    controller: [
+      AddMobileResourceController
+    ],
+    controllerAs: 'ctrl',
+    bindings: {
+      add: '<',
+      filterResources: '<',
+      filterExcluded: '<',
+      onClose: '<',
+      project: '<',
+      resource: '<',
+      resources: '<'
+    },
+    templateUrl: 'views/directives/add-mobile-resource.html'
+  });
+
+  function AddMobileResourceController() {
+    var ctrl = this;
+
+    ctrl.$doCheck = function () {
+      ctrl.excludedResources = ctrl.filterExcluded(ctrl.resource, ctrl.resources);
+    };
+
+    ctrl.addMobileResource = function (mobileResource) {
+      ctrl.add(mobileResource);
+      ctrl.onClose();
```

Comment:
Why `onClose()` called here

Reply:
Need to close the modal as soon as the resource is added (this will be removed now though when I replace the modal with the dropdown)

Reply:
Otherwise if not removed then call it `ctrl.addMobileResourceAndCloseModal`